pkg/auth/auth.go

```diff
@@ -219,16 +283,16 @@ func Logout(systemContext *types.SystemContext, opts *LogoutOptions, args []stri
 		if !opts.AcceptUnspecifiedRegistry {
 			return errors.New("please provide a registry to logout from")
 		}
-		if server, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
+		if key, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
 			return err
 		}
-		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", server)
+		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", key)
 	}
 	if len(args) != 0 {
 		if opts.All {
 			return errors.New("--all takes no arguments")
 		}
-		server = getRegistryName(args[0])
+		key = args[0]
```

Comment:
This should be parsing `args[0]` according to `opts.AcceptRepositories` so that we call `RemoveAuthentication` with the right `key`. Maybe like the `parseRegistryArgument` suggestion elsewhere.

Reply:
Yep, fixed as suggested.